extension/goext/error.go

```diff
@@ -15,43 +15,192 @@
 
 package goext
 
-import "fmt"
-
-// ErrorCode identifies an error code
-type ErrorCode int
+import (
+	"fmt"
+	"net/http"
+	"runtime"
+)
 
-// Error represents an error
+// Error represents an error code with related HTTP status
 type Error struct {
-	Code ErrorCode
-	Err  error
+	Err    error
+	Status int
+	Origin string
 }
 
 // NewError returns a new error
-func NewError(code ErrorCode, err error) Error {
-	return Error{
-		Code: code,
-		Err:  err,
+func NewError(status int, err error) *Error {
+	return &Error{
+		Err:    err,
+		Status: status,
+		Origin: captureOrigin(),
+	}
+}
+
+func captureOrigin() string {
+	const SkipStackFrames = 3
```

Comment:
please add a UT for this magic number

Reply:
Is there any case where we would like to see the full stack other than HandleEvent? My idea is changing this func to `func captureOrigin(skip int) string`, line 36 to `Origin: captureOrigin(3)` then add another constructor `func NewErrorFullStack(status int, err error) *Error`. What do you think?

Reply:
@p-kozlowski added UT
@morrisson I found current implementation to be very helpful when locating invalid error codes in code. In many cases, the same error code is returned from different places. Unless you have a working debugger for go extensions, it takes a lot of time to locate the exact "return Error" line which caused a particular error. In which places you would like to capture stack traces and in which this should be avoided in your opinion?

Reply:
Ok. This explanation makes sense to me.